+sw_tests/+unit_tests/unittest_spinw_spinwave.m

```diff
@@ -0,0 +1,471 @@
+            qpts = testCase.qh5;
+            sw_out = testCase.swobj.spinwave(qpts, 'fitmode', true);
+            % fitmode automatically turns off sortMode
+            expected_sw = testCase.swobj.spinwave(qpts, 'sortMode', false);
+            expected_sw = rmfield(expected_sw, {'obj', 'datestart', 'dateend'});
+            testCase.verify_spinwave(expected_sw, sw_out);
+        end
+        function test_symbolic(testCase)
+            % Test that spinw.spinwavesym() is called if spinw.symbolic==true and spinw.spinwave() is called
+            % Mock the spinwavesym and symbolic methods
+            [mocksw, bh] = testCase.createMock(?spinw, 'MockedMethods', {'symbolic', 'spinwavesym'});
+            % Make sure that spinw thinks it is symbolic
+            testCase.assignOutputsWhen(withAnyInputs(bh.symbolic), true);
+            % Make sure that spinwavesym outputs a sample spectra struct
+            hkl = [1 2; 3 4; 5 6];
+            out_spec = struct('hkl', hkl, 'swConv', [1; 2]);
+            testCase.assignOutputsWhen(withAnyInputs(bh.spinwavesym), out_spec);
+            % Call spinwave
+            spec = mocksw.spinwave(hkl);
+            testCase.assertCalled(withExactInputs(bh.spinwavesym()));
+            testCase.assertEqual(spec, out_spec);
+        end
+        function test_incommensurate(testCase)
+            % Tests that incommensurate calculation is ok
+            hkl = {[0 0 0] [0 1 0] [1 0 0] 5};
+            % Create copy to avoid changing obj for other tests
+            swobj = copy(testCase.swobj);
+            commensurate_spec = swobj.spinwave(hkl);
+            swobj.genmagstr('mode', 'direct', 'k', [0.123 0 0], 'n', [1 0 0], 'S', [0; 1; 0]);
+            % Forcing incomm struct means exchange parameters don't agree, so we need 'hermit' false
+            incomm_spec = swobj.spinwave(hkl, 'hermit', false);
+            testCase.assertEqual(size(incomm_spec.omega, 1), size(commensurate_spec.omega, 1) * 3);
+        end
+        function test_twin(testCase)
+            % Tests that setting twins gives correct outputs
+            % Create copy to avoid changing obj for other tests
+            swobj_twin = copy(testCase.swobj);
+            swobj_twin.addtwin('axis', [0 0 1], 'phid', [60 120], 'vol', [1 2]);
+            rotc = swobj_twin.twin.rotc;
+            hkl = [1 2; 3 4; 5 6];
+            sw_out = swobj_twin.spinwave(hkl);
+
+            expected_sw = testCase.default_spinwave;
+            expected_sw.param.notwin = false;
+            expected_sw.omega = {};
+            expected_sw.Sab = {};
+            expected_sw.obj = copy(swobj_twin);
+            expected_sw.hkl = hkl;
+            expected_sw.hklA = [2/3 4/3; 0.75, 1; 1.25, 1.5]*pi;
+            % Recalculate without twins for each set of hkl's and compare
+            [~, rotQ] = swobj_twin.twinq([0;0;0]);
+            for ii = 1:3
+                sw_single = testCase.swobj.spinwave((hkl' * rotQ(:, :, ii))');
```

Comment:
`twinq` will do the rotation of the Q points for each twin and return it in a cell as the first argument (rather than duplicate the logic here